### ISSUE TITLE
Add nvim-remote like functionality built in

### ISSUE THREAD

**Issue by golaneduard** [neovide/neovide#1137]
When i am trying to run neovide in terminal, without detaching, anyway are running in dettaching mode.
Command:
neovide --nofork /path/file
![image](https://user-images.githubusercontent.com/45820611/148663968-37e3b1f9-9a53-4334-a648-b5f10f5203c0.png)
Like this are showing when is with flag --nofork. Like to write something is not available.

I mean to run like vim or nvim, in terminal, without in external of terminal, if it's possible.

**Comment by fredizzimo** [comment]
I believe this issue is the same as https://github.com/neovide/neovide/issues/1122, which should be fixed once https://github.com/neovide/neovide/pull/1123 is merged.

**Comment by EduardGol** [comment]
> I believe this issue is the same as https://github.com/neovide/neovide/issues/1122, which should be fixed once https://github.com/neovide/neovide/pull/1123 is merged.
Thabks

**Comment by EduardGol** [comment]
> I believe this issue is the same as #1122, which should be fixed once #1123 is merged.

I have rebuild neovide with changes from #1123, and also not working how must to work with --nofork flag

**Comment by EduardGol** [comment]
> > I believe this issue is the same as #1122, which should be fixed once #1123 is merged.
> 
> I have rebuild neovide with changes from #1123, and also not working how must to work with --nofork flag

**Comment by EduardGol** [comment]
> I believe this issue is the same as #1122, which should be fixed once #1123 is merged.

I am ussing gnoem Terminal
![image](https://user-images.githubusercontent.com/45820611/148682926-68c76d69-9825-4446-8521-b5559d28e1b5.png)

**Comment by fredizzimo** [comment]
I haven't tested the changes myself, so I can't tell if they work or not and how they are supposed to work, but maybe @bpandreotti can answer.

**Comment by Uefair** [comment]
Neovide onto itself is a [GUI](https://en.wikipedia.org/wiki/Graphical_user_interface) application
the `--nofork` option stands for not detaching **process**, which is in charge for graphical window
in case you need terminal editing, just launch pure neovim, why would you want to launch a GUI app to launch a cli app?

i hope i got you right by what you meant `--nofork` flag to do

**Comment by EduardGol** [comment]
> Neovide onto itself is a [GUI](https://en.wikipedia.org/wiki/Graphical_user_interface) application
> the `--nofork` option stands for not detaching **process**, which is in charge for graphical window
> in case you need terminal editing, just launch pure neovim, why would you want to launch a GUI app to launch a cli app?
> 
> i hope i got you right by what you meant `--nofork` flag to do

The neovide windows are named like which file or folder has opened, like other application. After that, they are not gathered together, which give you much more comfortable to navigate between them.
Every time neovide are not opening new window of it self, but are opening new app like (neovide).
And if they will gather together , will be easier to switch with lonuz shortcut, for example ( alt+~).
Anyway, app is very nice, if not possible for moment no problem.

**Comment by Uefair** [comment]
tried making some sense of your sentences yet it makes nonsense:
>The neovide windows are named after the files or folders opened, like other
>applications. After that, they are not gathered together, which give you much
>more comfortable navigation between them.
>Every time neovide doesn't open new window of itself, but are opening new app like neovide.
>And if they gather together, it will be easier to switch with lonuz shortcut, for example super+~.

### Am i missing something?
![2022-01-09_19-01-1641745117](https://user-images.githubusercontent.com/62612309/148690919-38ba6a14-26e2-459a-9d57-69e1ee397e0d.png)

**Comment by EduardGol** [comment]
> tried making some sense of your sentences yet it makes nonsense:
> >The neovide windows are named after the files or folders opened, like other
> >applications. After that, they are not gathered together, which give you much
> >more comfortable navigation between them.
> >Every time neovide doesn't open new window of itself, but are opening new app like neovide.
> >And if they gather together, it will be easier to switch with lonuz shortcut, for example super+~.
> 
> ### Am i missing something?
> ![2022-01-09_19-01-1641745117](https://user-images.githubusercontent.com/62612309/148690919-38ba6a14-26e2-459a-9d57-69e1ee397e0d.png)
> 

Ok

**Comment by Kethku** [comment]
Just gonna chime in here. @Uefair I think saying somebody's well intentioned comment is nonesense isn't very productive and can be pretty demoralizing. I have no reason to think folks intended to be mean here but let's try to keep things positive.

@EduardGol can you try a build from https://github.com/neovide/neovide/pull/1123 to see if that fixes your issue?

**Comment by EduardGol** [comment]
> Just gonna chime in here. @Uefair I think saying somebody's well intentioned comment is nonesense isn't very productive and can be pretty demoralizing. I have no reason to think folks intended to be mean here but let's try to keep things positive.
> 
> @EduardGol can you try a build from https://github.com/neovide/neovide/pull/1123 to see if that fixes your issue?

All is good :).
What i must to do no, but what said @Uefair , about Gui is true, reason which i have wanted to run in terminal was few bugs, but with time i think they will be fixed with more reply and explication from users. Anyway is very nice app, with great future.

**Comment by fredizzimo** [comment]
I think he tried #1123 already and said that it didn't work. 

I'm also unsure what @EduardGol, actually means, there's clearly a small language barrier here. But, after reading his message again, I think he wants some kind of MDI or tabbed interface, where everything is openened in a single window. So the bug/feature request does not actually have anything to do with the `-nofork` flag, which has an entirely different purpose.

That can currently be done with https://github.com/mhinz/neovim-remote, but then you need to launch nvim as a TCP server. So it's not very user friendly, 

So I think the feature request is for Neovide to include some functionality to detect that it's running, and then call Neovim with RPC to open a new document instead of launching a new instance.

**Comment by Kethku** [comment]
my bad...

**Comment by Kethku** [comment]
This is an interesting idea. We could have neovide setup its own listening server and build neovim-remote like functionality in to send and receive neovim commands. This would make it automatic and not require python or nvr to work...

That said it seems like a pretty big design surface area that I'd have to think about more to build a PR for. I'm interested in your thoughts.

As a side note, neovim-remote has some issues for sure, but its my goto and I currently have a powershell alias for using it

```
function e
{
  param(
    $fileName = $null
  )
  if ($env:NVIM_LISTEN_ADDRESS) {
    nvr -cc "call v:lua.g.hide_terminal()" $fileName
  } else {
    neovide $fileName
  }
}
```

This lets me use `e` to edit a file. When in a neovim terminal, this opens the file in a new buffer in the existing window, but when in another app, it starts a new instance of neovide and edits it there. Something like this which looks for an existing neovide window and opens there automatically seems reasonable and generally useful.

Any thoughts?

**Comment by fredizzimo** [comment]
Yes, exactly, I don't have much more to add, I haven't thought about it either. But most editors works this way, at least on Windows. Usually they don't have many other options than to just open in the last active instance, if there are one or more instances open, and if no instance is open, then launch a new one.

Also related, but a different thing is integration with the operating system `open with` menu, which should use the same functionality.

**Comment by Kethku** [comment]
I renamed this issue and gave it the enhancement tag as this is an improvement now and not a bug.

**Comment by Kethku** [comment]
We support connecting to remote servers and neovim supports nvim-remote now. This is no longer relevant.
